### DISCUSSION TITLE
Which function or code is for calculating the highlighting parts for two strings?

### DISCUSSION THREAD

**Discussion by roadfar** [dandavison/delta#1913] [General]
As lines 5, 6, 13, and 23 show in the following figure show, which function or code is for calculating the highlighting parts (i.e., the string start index and end index) of the left and right lines?
![image](https://github.com/user-attachments/assets/e2066e16-f46c-472c-8275-0750a9b98267)

**Comment by th1000s** [comment]
That is done at `infer_edits(..)` in [src/edits.rs:24](https://github.com/dandavison/delta/blob/d10095bc732a767/src/edits.rs#L24). Plus there is [--color-moved](https://dandavison.github.io/delta/color-moved-support.html) support, if git provides this information.
